sycl/doc/extensions/SPIRV/SPV_INTEL_memory_access_aliasing.asciidoc

```diff
@@ -0,0 +1,275 @@
+Revision 5.
+
+This extension requires SPIR-V 1.0.
+
+Overview
+--------
+
+This extension adds *AliasDomainDeclINTEL*, *AliasScopeINTEL* and
+*AliasScopeListDeclINTEL* instructions that serve a purpose to
+specify aliasing scopes within which annotated memory accesses don't alias.
+Results of these instructios are used by *AliasScopeINTEL* and *NoAliasINTEL*
+decorations and Memory Operands mask to annotate memory access instructions.
+
+Extension Name
+--------------
+
+To use this extension within a SPIR-V module, the following *OpExtension* must
+be present in the module:
+
+----
+OpExtension "SPV_INTEL_memory_access_aliasing"
+----
+
+New Capabilities
+----------------
+This extension introduces a new capability:
+
+----
+MemoryAccessAliasingINTEL
+----
+
+New instructions
+----------------
+
+Instructions added under the *MemoryAccessAliasingINTEL* capability:
+
+----
+AliasDomainDeclINTEL
+AliasScopeDeclINTEL
+AliasScopeListDeclINTEL
+----
+
+New Decorations
+---------------
+Decorations added under the *MemoryAccessAliasingINTEL* capability:
+
+----
+AliasScopeINTEL
+NoAliasINTEL
+----
+
+Token Number Assignments
+------------------------
+[width="45%",cols="30,15"]
+|=================================
+| MemoryAccessAliasingINTEL | 5910
+| AliasDomainDeclINTEL      | 5911
+| AliasScopeDeclINTEL       | 5912
+| AliasScopeListDeclINTEL   | 5913
+| AliasScopeINTEL           | 5914
+| NoAliasINTEL              | 5915
+|=================================
+
+Modifications to the SPIR-V Specification, Version 1.5
+------------------------------------------------------
+
+Modifications 2.4 Logical Layout of a Module
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Insert an extra item before "8. All annotation instructions"
+
+8. Optional _AliasDomainDeclINTEL_, _AliasScopeDeclINTEL_ and _AliasScopeDeclINTEL_
+instructions.
+
+
+Capabilities
+~~~~~~~~~~~~
+
+Modify Section 3.31, "Capability", adding these rows to the Capability table:
+
+--
+[options="header"]
+|====
+2+^| Capability ^| Depends On
+| 5910 | *MemoryAccessAliasingINTEL* +
+Allow to use AliasDomainDeclINTEL, AliasScopeDeclINTEL and
+AliasScopeListDeclINTEL instructions, AliasScopeINTEL and NoAliasINTEL and
+AliasScopeMaskINTEL and NoAliasMaskINTEL memory operands.
+decorations |
+|====
+--
+
+Instructions
+~~~~~~~~~~~~
+
+[cols="4", width="100%"]
+|=====
+3+^|*OpAliasDomainDeclINTEL* +
+The instruction declares an aliasing domain.
+
+The _Result_ is used by _AliasScopeDeclINTEL_ instruction to form aliasing
+scope. When evaluating an aliasing query if two memory access instructions that
+have _AliasScopeMaskINTEL_ and _NoAliasMaskINTEL_ memory operands or decorated
+with _AliasScopeINTEL_ and _NoAliasINTEL_ decorations and if for some domain,
+the set of scopes with that domain in one instruction’s aliasing scope list is a
+subset of (or equal to) the set of scopes for that domain in another
+instruction’s noalias list, then the two memory accesses are assumed not to
+alias.
```

Comment:
I have moved this text to overview and explicitly specify "Or one has the first one and another the second one?".

Reply:
>> Does this spelling assume that those memory instructions have both of those operands at the same time? Or one has the first one and another the second one?

Yes, a single instruction can have both memory operands simultaneously in case if they have different scope, like
```
%var = OpLoad %type %ptr1 (noalias 1, alias 2)
OpStore %ptr2 %var (noalias 1)
OpStore %ptr3 %var (alias 2)
```